sync/sync.go

```diff
@@ -627,55 +614,233 @@ func (s *Synchronizer) fetchAndStorePending(ctx context.Context) error {
 	return s.StorePending(&pending)
 }
 
-func (s *Synchronizer) fetchAndStorePreConfirmed(ctx context.Context) error {
-	highestBlockHeader := s.highestBlockHeader.Load()
-	if highestBlockHeader == nil {
-		return nil
+// pollPreLatest fetches preLatest block for each new head and forwards the prelatest to given chanel.
+//
+// Consumer of preLatestChan never receives the same preLatest twice.
```

Comment:
On the same idea as the previous comment:
```suggestion
// pollPreLatest fetches the pre-latest block for each new head and forwards it to a given 
// channel (`preLatesChan`).
// Consumers of `preLatestChan` will never receive the same pre-latest block twice. 
```